mousegrabber.c

```diff
@@ -98,23 +99,31 @@ luaA_mousegrabber_run(lua_State *L)
     if(globalconf.mousegrabber != LUA_REFNIL)
         luaL_error(L, "mousegrabber already running");
 
-    uint16_t cfont = xcursor_font_fromstr(luaL_checkstring(L, 2));
+    uint16_t cfont;
 
-    if(cfont)
+    if(!lua_isnil(L, 2))
     {
-        xcb_cursor_t cursor = xcursor_new(globalconf.cursor_ctx, cfont);
-
-        luaA_registerfct(L, 1, &globalconf.mousegrabber);
-
-        if(!mousegrabber_grab(cursor))
-        {
-            luaA_unregister(L, &globalconf.mousegrabber);
-            luaL_error(L, "unable to grab mouse pointer");
-        }
+        cfont = xcursor_font_fromstr(luaL_checkstring(L, 2));
+        if(!cfont)
+            luaA_warn(L, "invalid cursor");
     }
     else
-        luaA_warn(L, "invalid cursor");
+        cfont = XCB_NONE;
 
+    xcb_cursor_t cursor;
+    if(cfont)
+        cursor = xcursor_new(globalconf.cursor_ctx, cfont);
+    else
+        cursor = XCB_NONE;
```

Comment:
This still alters the behaviour. Previously, nothing happened when an invalid value was passed. Now, it defaults to `XCB_NONE`.
And the whole bit can be simplified.

The two combined:
```c
    xcb_cursor_t cursor = XCB_NONE;

    if(!lua_isnil(L, 2))
    {
        uint16_t cfont = xcursor_font_fromstr(luaL_checkstring(L, 2));
        if(!cfont)
        {
            luaA_warn(L, "invalid cursor");
            return 0;
        }

        cursor = xcursor_new(globalconf.cursor_ctx, cfont);
    }
```